projects/dex-ui/src/wells/useWellHistory.tsx

```diff
@@ -0,0 +1,115 @@
+import { useQuery } from "@tanstack/react-query";
+import { GetWellEventsDocument, GetWellEventsQuery, Token } from "src/generated/graphql";
+
+import { TokenValue } from "@beanstalk/sdk";
+import { fetchFromSubgraphRequest } from "./subgraphFetch";
+
+export enum EVENT_TYPE {
+  SWAP,
+  ADD_LIQUIDITY,
+  REMOVE_LIQUIDITY
+}
+
+export type WellEvent = {
+  type: EVENT_TYPE;
+  hash: string;
+  totalDollarValue: number;
+  label: string;
+  timestamp: number;
+};
+
+// TODO: Store in env var?
+const HISTORY_DAYS = 7;
+
+const useWellHistory = (wellId: string) => {
+  const HISTORY_DAYS_AGO_BLOCK_TIMESTAMP = Math.floor(new Date(Date.now() - HISTORY_DAYS * 24 * 60 * 60 * 1000).getTime() / 1000);
+
+  const fetcher = async () => {
+    console.debug(`Fetching WellEvents for last ${HISTORY_DAYS_AGO_BLOCK_TIMESTAMP} days`);
+    const data = await fetchFromSubgraphRequest(GetWellEventsDocument, {
+      id: wellId,
+      searchTimestamp: HISTORY_DAYS_AGO_BLOCK_TIMESTAMP
+    });
+
+    const results = await data();
+
+    let swaps: WellEvent[] = [];
+    if (results.well?.swaps) {
+      swaps = handleSwapEvents(results);
+    }
+
+    let deposits: WellEvent[] = [];
+    if (results.well?.deposits) {
+      deposits = handleLiquidityEvents(EVENT_TYPE.ADD_LIQUIDITY, results);
+    }
+
+    let withdraws: WellEvent[] = [];
+    if (results.well?.withdraws) {
+      withdraws = handleLiquidityEvents(EVENT_TYPE.REMOVE_LIQUIDITY, results);
+    }
+
+    const allData = swaps.concat(deposits).concat(withdraws);
+    return allData.sort((a, b) => b.timestamp - a.timestamp);
+  };
+
+  const getTokenHumanAmount = (decimals: number, amount: string) =>
+    parseFloat(TokenValue.fromBlockchain(amount, decimals).toHuman()).toFixed(0);
+
+  const generateLiquidityLabel = (tokens: Token[], reserves: string[]) => {
+    const labels: string[] = [];
+    tokens.forEach((token, index) => {
+      labels.push(`${getTokenHumanAmount(token.decimals, reserves[index])} ${token.symbol}`);
+    });
+    return labels.join(" AND ");
+  };
+
+  const calculateSwapUSDValue = (amountOut: string, toToken: Token) => {
+    const _amountOut = parseFloat(TokenValue.fromBlockchain(amountOut, toToken.decimals).toHuman());
+    const _price = parseFloat(toToken.lastPriceUSD);
+    const toTokenValue = _price * _amountOut;
+    return toTokenValue;
+  };
+
+  const getAmountForLabel = (amount: string, token: Token) =>
+    parseFloat(TokenValue.fromBlockchain(amount, token.decimals).toHuman()).toFixed(0);
```

Comment:
toHuman(format) can format things for you now. You can use whatever [format](http://numeraljs.com/#format)

```suggestion
  const getAmountForLabel = (amount: string, token: Token) => TokenValue.fromBlockchain(amount, token.decimals).toHuman('0.0a);

```

Reply:
Is this change in `wells-main` or another branch?